lib/ramble/docs/tutorials/modifying_a_gromacs_experiment.rst

```diff
@@ -0,0 +1,221 @@
+      Application: gromacs
+        Workload: water_bare
+          Experiment: gromacs.water_bare.rf_single_rank
+            Variables from Workspace:
+              processes_per_node = 16 ==> 16
+              mpi_command = mpirun -n {n_ranks} -ppn {processes_per_node} ==> mpirun -n 1 -ppn 16
+              batch_submit = {execute_experiment} ==> {execute_experiment}
+            Variables from Experiment:
+              n_ranks = 1 ==> 1
+              n_threads = 1 ==> 1
+              size = 0003 ==> 0003
+              type = rf ==> rf
+
+The variables ``mpi_command``, ``processes_per_node``, and ``batch_submit``
+come from the workspace scope on lines 3, 4, and 5 of the YAML file in
+:ref:`gromacs_workspace_config`. Each experiment has its own definition for
+``n_ranks``, ``n_threads``, ``size``, and ``type``.
+
+The experiments in this configuration use both the ``water_bare`` and
+``water_gmx50`` workloads.
+
+Using Workload Variables
+------------------------
+
+Application definitions are allowed to expose variables that can be used within
+experiments. These are called **workload variables**. User can view the
+available variables with the following command:
+
+.. code-block:: console
+
+    $ ramble info gromacs
+
+
+Focusing on the relevant workloads, you see the following information:
+
+.. code-block:: console
+
+    Workload: water_gmx50
+        Executables: ['builtin::env_vars', 'builtin::spack_source', 'builtin::spack_activate', 'pre-process', 'execute-gen']
+        Inputs: ['water_gmx50_bare']
+        Variables:
+            size:
+                Description: Workload size
+                Default: 1536
+                Suggested Values: ['0000.65', '0000.96', '0001.5', '0003', '0006', '0012', '0024', '0048', '0096', '0192', '0384', '0768', '1536', '3072']
+            type:
+                Description: Workload type.
+                Default: pme
+                Suggested Values: ['pme', 'rf']
+            input_path:
+                Description: Input path for water GMX50
+                Default: {water_gmx50_bare}/{size}
+    Workload: water_bare
+        Executables: ['builtin::env_vars', 'builtin::spack_source', 'builtin::spack_activate', 'pre-process', 'execute-gen']
+        Inputs: ['water_bare_hbonds']
+        Variables:
+            size:
+                Description: Workload size
+                Default: 1536
+                Suggested Values: ['0000.65', '0000.96', '0001.5', '0003', '0006', '0012', '0024', '0048', '0096', '0192', '0384', '0768', '1536', '3072']
+            type:
+                Description: Workload type.
+                Default: pme
+                Suggested Values: ['pme', 'rf']
+            input_path:
+                Description: Input path for water bare hbonds
+                Default: {water_bare_hbonds}/{size}
+
+Using this information, change the size on each experiment based on the
+suggested values. **NOTE** The larger the value, the more expensive the
```

Comment:
Do we want to link to a file with the edits done for the user, so they can click and compare they did it right at the end?

Reply:
Also, bit of an annoying nit, but this tutorial about modify is >200 lines but this one line of "ok edit the thing!" is kind of subtle and hidden. Do we want to make it more obvious some how? I feel like people might skim over it and miss the point

Reply:
I don't think I want to link to a file with the edits made. Mostly because I want the people going through the tutorial to pick a size and put it into the configuration file (which means it's harder to give a 'true' answer since there are like 9 different values they can use).

But yes, let me see if I can bring more emphasis to the editing part.

Reply:
OK, this is updated. PTAL